google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -118,6 +118,90 @@ def table_ref
           table_ref
         end
 
+        ###
+        # Is the table partitioned?
+        #
+        # @!group Attributes
+        #
+        def time_partitioning?
+          !@gapi.time_partitioning.nil?
+        end
+
+        ###
+        # The period for which the table is partitioned, if any.
+        #
+        # @!group Attributes
+        #
+        def time_partitioning_type
+          ensure_full_data!
+          @gapi.time_partitioning.type if time_partitioning?
+        end
+
+        ##
+        # Sets the partitioning for the table. See [Partitioned Tables
+        # ](https://cloud.google.com/bigquery/docs/partitioned-tables).
+        #
+        # @param [String] type The partition type. Currently the only
+        # supported value is "DAY".
+        #
+        # @example
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #   dataset = bigquery.dataset "my_dataset"
+        #   table = dataset.create_table "my_table" do |table|
+        #     table.time_partitioning_type = "DAY"
+        #   end
+        #
+        # @!group Attributes
+        #
+        def time_partitioning_type= type
+          @gapi.time_partitioning ||=
+              Google::Apis::BigqueryV2::TimePartitioning.new
+          @gapi.time_partitioning.type = type
+          patch_gapi! :time_partitioning
+        end
+
+
+        ###
+        # The expiration for the table partitions, if any,  in milliseconds.
```

Comment:
@blowmage Yes, well, almost always. :) See my comments in the PR about `Dataset#default_expiration` and @quartzmo's comments in #1591.

> Hmmm, although we generally present seconds, it looks like Dataset#default_expiration preserves the milliseconds of the underlying API. I suppose we will have to debate this in your PR.

> I'd say use milliseconds for now, and if there is a consensus that everything should be in seconds, then we'll have to discuss changing Dataset#default_expiration as well.

So let's open the debate. :)